### DISCUSSION TITLE
How to make a one-page-layout?

### DISCUSSION THREAD

**Discussion by MrMacStripe** [datenstrom/community#181] [Ask a question] [answered]
First of all, thanks for the amazing system, I love it and used it in many projects.

At the moment I´m trying to realize a one-page-layout. My goal is to have multiple sections that use different page.text files OR one page.txt file with sections inside that will appear in given positions. My problem is, I don´t get my head around how to use the API. To make it clear, I want a single page:

home
news
page-1
page-2
page-3

Home and news are working. Now I need to load snippets of a page.txt into divs. In a perfect case you could edit the content with the web application. 

How can I achieve that? Thanks in advance :)

**Comment by markseuffert** [comment]
Good idea, it should be possible to make a one-pager. Here's a starting point: Write your own template and then you can access any page with the [Yellow API](https://datenstrom.se/yellow/help/api-for-developers). For example `$yellow->page->getChildren()` gives you access to pages in the same folder.

**Comment by MrMacStripe** [comment, marked as answer]
Okay, for the moment I use this code:

```
<section id="termine" class="row align-center">

<?php list($name, $location, $pagesMax) = $yellow->getSnippetArgs(); ?>
<?php if(empty($location)) $location = "/termine/" ?>
<!--<?php if(empty($pagesMax)) $pagesMax = 3; ?>-->
<?php $termine = $yellow->pages->find($location); ?>
<?php $pages = $termine ? $termine->getChildren() : $yellow->pages->clean(); ?>
<!--<?php $pages->limit($pagesMax); ?>-->

    <?php foreach($pages as $page): ?>
     <div class="entry-text columns">
        <div class="entry-header"><h3><a href="<?php echo $page->getLocation() ?>"><?php echo $page->getHtml("title") ?></a></h3></div>
        <div class="entry-content"><?php echo $yellow->page->getContent() ?></div>
    </div>
    <?php endforeach ?>
</section>
```

Please note that this is part of my site. This should work for me I guess. Had troubles with the API yesterday, but after some sleep I found my problem immediately.

**Comment by markseuffert** [comment]
Thanks for sharing your code.

Btw, if you didn't add it somewhere else in your code I suggest to update the modified date. When a page is created from multiple files, it's a good idea to update the last modified date, which typically is used for [caching](https://www.mnot.net/cache_docs/). This let's the web browser know when the content was modified.  For example in your code:

```
...
<?php $pages = $termine ? $termine->getChildren() : $yellow->pages->clean(); ?>
<?php $yellow->page->setLastModified($pages->getModified()) ?>
...
```

Hope everything works as expected. Feedback is much appreciated. 😄

**Comment by MrMacStripe** [comment]
Thanks for the hint. I´ll share my ideas and my code once I finished the site that´s due tomorrow evening :D - really busy completing the task ;)

I guess there´s something I´m missing in terms of the general way how yellow displays and  creates content. But that is for discussion once I can share my code...

**Comment by markseuffert** [comment]
Feel free to ask questions. Would be nice to get more developers on board...
